kratos/spatial_containers/geometrical_objects_bins.h

```diff
@@ -222,6 +222,47 @@ class KRATOS_API(KRATOS_CORE) GeometricalObjectsBins
         return results;
     }
 
+    /**
+     * @brief This method takes a point and finds all of the objects in the given bounding box to it.
+     * @details The result contains the object and also its distance to the point.
+     * @param rPoint The point to be checked
+     * @param rMinPoint The minimum point of the bounding box
+     * @param rMaxPoint The maximum point of the bounding box
+     * @param rResults The results of the search
+     */
+    void SearchInBoundingBox(
+        const PointType& rPoint,
+        const array_1d<double, 3>& rMinPoint,
+        const array_1d<double, 3>& rMaxPoint,
+        std::vector<ResultType>& rResults
+        );
+
+    /**
+     * @brief This method takes a point and finds all of the objects in the given bounding box to it (iterative version).
+     * @details The result contains the object and also its distance to the point.
+     * @param itPointBegin The first point iterator
+     * @param itPointEnd The last point iterator
+     * @param rMinPoint The minimum point of the bounding box
+     * @param rMaxPoint The maximum point of the bounding box
+     * @param rResults The results of the search
+     * @tparam TPointIteratorType The type of the point iterator
+     */
+    template<typename TPointIteratorType>
```

Comment:
I don't understand what this method does? The result should not depend on the point!

Reply:
I don't understand how do you search without a point to be searched